crates/ide/src/highlight_related.rs

```diff
@@ -232,6 +232,22 @@ fn highlight_references(
             }
         }
 
+        if matches!(def, Definition::Label(_)) {
+            let label = token.parent_ancestors().find_map(ast::Label::cast);
```

Comment:
```suggestion
            let label = token.parent().and_then(ast::Label::cast);
```

Reply:
The hierarchy of an `ast::Label` is structured as `(Label (Lifetime ident) colon)`, which suggests that the `token` here may refer to the `ident` within `Lifetime`.

If we don't use `token.parent_ancestors`, we should write:

```rust
let label = token.parent().and_then(|token| {
    ast::Label::cast(token.clone())
        .or_else(|| token.parent().and_then(ast::Label::cast))
});
```

... which is more complicated 🤔

Reply:
Ah right, lifetime is a node as well, then lets do
```suggestion
let label = token.parent_ancestors().nth(1).and_then(ast::Label::cast);
```